polkadot-protocol/protocol-operations/consensus-protocols.md

```diff
@@ -0,0 +1,109 @@
+
+When multiple validators are block producer candidates in a given slot, all will produce a block and broadcast it to the network. At that point, it becomes a race to reach most of the network first. Depending on network topology and latency, both chains will continue to build in some capacity until finalization kicks in. The [Fork Choice](#fork-choice) rule determines which chain to follow.
+
+#### No Validators in Slot
+
+To handle potentially empty slots when no validators have rolled low enough in the randomness lottery to qualify for block production, a [secondary validator selection algorithm](https://spec.polkadot.network/sect-block-production#defn-babe-secondary-slots){target=\_blank} runs in the background. The validators selected through this predictable algorithm always produce blocks. These secondary blocks are ignored if the same slot has a primary block produced from a [VRF-selected](https://wiki.polkadot.network/docs/learn-cryptography#vrf){target=\_blank} validator. As a result, a slot can have either a primary or a secondary block, and no slots are ever skipped.
+
+!!!note
+    For more detailed information on BABE, including its cryptographic underpinnings and formal proofs, please refer to the [BABE paper](https://research.web3.foundation/Polkadot/protocols/block-production/Babe){target=\_blank}.
+
+## Finality Gadget: GRANDPA
+
+GRANDPA (GHOST-based Recursive ANcestor Deriving Prefix Agreement) is the finality gadget implemented for the Polkadot relay chain. It works in parallel to the BABE block production mechanism to provide provable finality.
+
+Key features of GRANDPA include:
+
+- Runs as an independent service parallel to block production
+- Reaches agreements on chains rather than individual blocks
+- Can finalize multiple blocks at once, greatly speeding up the finalization process
+- Operates in a partially synchronous network model
+- Can cope with up to 1/5 Byzantine nodes in an asynchronous setting
+
+### Probabilistic vs. Provable Finality
+
+A pure Nakamoto consensus blockchain that runs PoW is only able to achieve the notion of probabilistic finality and reach eventual consensus. Probabilistic finality means that under some assumptions about the network and participants, if we see a few blocks building on a given block, we can estimate the probability that it is final. Eventual consensus means that at some point in the future, all nodes will agree on the truthfulness of one set of data. This eventual consensus may take a long time, and will not be able to determine how long it will take ahead of time. However, finality gadgets such as GRANDPA (GHOST-based Recursive ANcestor Deriving Prefix Agreement) or Ethereum's Casper FFG (the Friendly Finality Gadget) are designed to give stronger and quicker guarantees on the finality of blocks - specifically, that they can never be reverted after some process of Byzantine agreements has taken place. The notion of irreversible consensus is known as provable finality.
```

Comment:
Grammarly suggestion

```suggestion
A pure Nakamoto consensus blockchain that runs PoW can only achieve probabilistic finality and eventual consensus. Probabilistic finality means that under some assumptions about the network and participants, we can estimate the probability that it is final if we see a few blocks building on a given block. Eventual consensus means that at some point in the future, all nodes will agree on the truthfulness of one set of data. This eventual consensus may take a long time, and will not be able to determine how long it will take ahead of time. However, finality gadgets such as GRANDPA (GHOST-based Recursive ANcestor Deriving Prefix Agreement) or Ethereum's Casper FFG (the Friendly Finality Gadget) are designed to give more robust and quicker guarantees on the finality of blocks - specifically, that they can never be reverted after some process of Byzantine agreements has taken place. The notion of irreversible consensus is known as provable finality.
```